### ISSUE TITLE
JSONPath evaluation

### ISSUE THREAD

**Issue by tschmidtb51** [csaf-poc/csaf_distribution#299]
The following items in the category config with valid JSON Path values are not evaluated correctly:
- `"expr:$.product_tree..branches[?(@.category=='vendor' || @.category=='product_family' || @.category=='product_name' || @.category=='product_version')].name"`
- `"expr:$.vulnerabilities[*].cve"`
- `"expr:$.vulnerabilities[*].cwe.id"`

**Comment by s-l-teichmann** [comment]
The `[*]` terms let the expressions result in arrays and not in single strings.
E.g. applying `$.vulnerabilities[*].cve`  to
```json
{ "vulnerabilities": [                                                                                                                                     
    { "cve": "foo" },                                                                                                                                      
    { "cve": "bar" }                                                                                                                                       
]}
```
will result in
``` json
[
  "foo",
  "bar"
]
```

Using `$.vulnerabilities[0].cve` should work as it is extracting only the first item `foo`.

**Comment by tschmidtb51** [comment]
I guess I was just reading what I was looking for - not what was written in the documentation. 

Anyway, I think we should support these JSON path values. (Otherwise, I would have to specify many single expressions if I look at the [recent Oracle CSAF](https://www.oracle.com/docs/tech/security-alerts/cpujan2023csaf.json).)

**Comment by bernhardreiter** [comment]
Add to documentation as example default configuration:
```
"expr:$.vulnerabilities[*].cve"
"expr:$.vulnerabilities[*].cwe.id"
```

and where failure is documented (or and improve log message).

Entries should be unique (enforced).

**Comment by s-l-teichmann** [comment]
This is addressed by PR #307
